libs/SalesforceSDK/src/com/salesforce/androidsdk/accounts/UserAccountManager.java

```diff
@@ -73,6 +74,50 @@
 
 	public static final String USER_SWITCH_INTENT_ACTION = "com.salesforce.USERSWITCHED";
 
+	/**
+	 * Found in an intent with action {@link #USER_SWITCH_INTENT_ACTION}, this extra specifies how a user is switching.
+	 * {@link UserSwitchType} can be used to annotate extra values retrieved from an intent's bundle.
+	 *
+	 * @see UserSwitchType
+	 */
+	public static final String EXTRA_USER_SWITCH_TYPE = "com.salesforce.USER_SWITCH_TYPE";
+
+	/**
+	 * Represents how the current user has been switched to, as found in an intent sent to a {@link android.content.BroadcastReceiver}
+	 * filtering {@link #USER_SWITCH_INTENT_ACTION}. User switching including logging in, logging out and switching between authenticated
+	 * users. For backwards compatibility, the case where the last user has logged out is not included, as this currently does not
+	 * send a broadcast.
+	 *
+	 * @author cliban
+	 */
+	@IntDef({
```

Comment:
Let me know if the MSDK is supporting IntDef. It's not required for functionality but is nice to have.